src/main/java/in/ekstep/am/step/TokenSignStep.java

```diff
@@ -0,0 +1,132 @@
+package in.ekstep.am.step;
+
+import in.ekstep.am.dto.token.TokenSignRequest;
+import in.ekstep.am.jwt.*;
+import in.ekstep.am.builder.TokenSignResponseBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.text.MessageFormat.format;
+
+public class TokenSignStep implements TokenStep {
+
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    @Autowired
+    TokenSignResponseBuilder tokenSignResponseBuilder;
+    @Autowired
+    private KeyManager keyManager;
+    @Autowired
+    TokenSignRequest token;
+
+    private String currentToken, kid, iss;
+    private Map headerData, bodyData;
+    private KeyData keyData;
+    private long tokenValidity, offlineTokenValidity, iat, refreshIat;
+
+    public TokenSignStep(TokenSignRequest token, TokenSignResponseBuilder tokenSignResponseBuilder, KeyManager keyManager) {
+        this.token = token;
+        this.tokenSignResponseBuilder = tokenSignResponseBuilder;
+        this.keyManager = keyManager;
+    }
+
+    private boolean isJwtTokenValid() {
+        currentToken = token.getRefresh_token();
+        headerData = GsonUtil.fromJson(new String(Base64Util.decode(currentToken.split("\\.")[0], 11)), Map.class);
+        bodyData = GsonUtil.fromJson(new String(Base64Util.decode(currentToken.split("\\.")[1], 11)), Map.class);
+
+
+        if(currentToken.split("\\.").length != 3){
+            log.error("Invalid JWT token: " + currentToken);
+            return false;
+        }
+
+        kid = keyManager.getValueFromKeyMetaData("token.kid");
+        if (!headerData.get("kid").equals(kid)) {
+            log.error("Invalid kid: " + kid);
```

Comment:
- Change all to log.error from log.info
- Seprate validation and generation blocks